### PR TITLE
Design Preview now supports Primary, Secondary color settings

### PR DESCRIPTION
**Pull request by kjohnson**
## Description

This PR introduces the `primaryColor` and `secondaryColor` settings used by the selected template.

## Affects

- `StorageDriver` `preview` method now accepts `FormData`, which includes form settings as well as block data.
- `DonationFormViewRouteData` and `DonationFormViewController` now support the primary color as a custom CSS property on the application container. 

## Visuals

Peek 2022-11-10 11-54

image

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by kjohnson** [comment]
@jonwaldstein this is finally ready for review. It took a while to un-tagle how to adjust to placement of each popover so that they do not overlap with our custom sidebar.

File: packages/form-builder/src/types/formSettings.ts [hunk 1]
```diff
@@ -8,4 +8,6 @@ export type FormSettings = {
     registration: string;
     goalFormat: string;
     templateId: string;
+    primaryColor: string;
+    secondaryColor: string;
```
Comment (jonwaldstein) [inline packages/form-builder/src/types/formSettings.ts:12]: Let's add the GiveWP branding colors as the defaults for these in `App.tsx` within initialState.  If these are empty the template is going to look very strange.

File: src/NextGen/DonationForm/DataTransferObjects/DonationFormViewRouteData.php [hunk 2]
```diff
@@ -32,7 +34,7 @@ public static function fromRequest(array $request): DonationFormViewRouteData
         $self = new static();
 
         $self->formId = (int)$request['form-id'];
-        $self->formTemplateId = !empty($request['form-template-id']) ? $request['form-template-id'] : '';
+        $self->formSettings = !empty($request['form-settings']) ? $request['form-settings'] : [];
```
Comment (jonwaldstein) [inline src/NextGen/DonationForm/DataTransferObjects/DonationFormViewRouteData.php:37]: @kjohnson I was hesitant about adding an array to the query params but I think it's okay. Just remember we are using this both as a post and get request for the form builder and the block rendering so I went ahead and fixed the latter 6161e622

This is what that iframe url will look like going this route:
```
/?givewp-view=donation-form&form-id=2658&form-settings[formTitle]=Test&form-settings[enableDonationGoal]=0&form-settings[enableAutoClose]=0&form-settings[registration]=none&form-settings[goalFormat]=amount-raised&form-settings[templateId]=classic
```

File: src/NextGen/DonationForm/Controllers/DonationFormViewController.php [hunk 3]
```diff
@@ -32,7 +33,10 @@ public function show(DonationFormViewRouteData $data): string
         $viewModel = new DonationFormViewModel($data->formId, $data->formBlocks);
 
         wp_enqueue_global_styles();
-        $this->enqueueFormScripts($data->formId, $data->formTemplateId);
+        $this->enqueueFormScripts($data->formId, $data->formSettings['templateId']);
+
+        $primaryColor = $data->formSettings['primaryColor'];
+        $secondaryColor = $data->formSettings['secondaryColor'];
```
Comment (jonwaldstein) [inline src/NextGen/DonationForm/Controllers/DonationFormViewController.php:39]: Let's add a nullish condition for these to display a string if null. I like having the flexibility of piecemealing the form settings when it comes to the iframe url arguments.

**Review by jonwaldstein — CHANGES_REQUESTED** [review]
Just some minor notes but overall looking great 👍

**Comment by jonwaldstein** [comment]
@kjohnson this just came to me tonight and wanted to get this update in.  when rendering the donation form on the front-end, we want all the settings to come from the donation model _except_ specifically overriding them (which would be _either_ from the form builder design preview _or_ a donation form block setting - down the road).  

2ed229d9

File: packages/form-builder/src/types/formSettings.ts (hunk 1 above)
Comment (kjohnson) [reply packages/form-builder/src/types/formSettings.ts:12]: Resolved by https://github.com/impress-org/givewp-next-gen/pull/87/commits/52496c8e4f37b35d767dc56f1e0191d13e007a02.

File: src/NextGen/DonationForm/DataTransferObjects/DonationFormViewRouteData.php (hunk 2 above)
Comment (kjohnson) [reply src/NextGen/DonationForm/DataTransferObjects/DonationFormViewRouteData.php:37]: Yeah, I had considered encoding the settings, but the flat array works out well with the default array behavior. We can adjust it if it becomes a problem.

**Review by jonwaldstein — APPROVED** [review]
This is awesome! 🎨

**Pull request merged**